lib/dynamodb/document_client.d.ts

```diff
@@ -95,7 +95,7 @@ export namespace DocumentClient {
         /**
          * An optional pre-configured instance of the AWS.DynamoDB service object to use for requests. The object may bound parameters used by the document client.
          */
-        service?: DynamoDB
+        service?: DynamoDB<Params>
```

Comment:
I know the `Params` is required here but why does it have any thing to do with the typings?

Reply:
I should probably verify this in the code but my assumption is that if you bind say the `TableName` parameter in the `DynamoDB` constructor and then pass it as the service to the `DynamoDB.DocumentClient` constructor you should be able to omit the `TableName` parameter. This generic parameter here helps with Type Inference, so that you don't need to actually specify the type of `Params` anywhere....

basically this:
```
const client = new DynamoDB.DocumentClient({
  service: new DynamoDB({
   params: {
     TableName: 'MyTable'
    }
  })
});
// can omit TableName
client.get({Key: {id: '1234'}}, console.log)
```
I added a case to the Dynamo typescript file (the one that validates the types) to show that this was working but now that I type this I think I want to also validate the case that I can pass both a service with parameters and a additional parameters and all of them should be omitted from the method calls.